src/Microsoft.Diagnostics.Monitoring.RestServer/Controllers/DiagController.cs

```diff
@@ -75,26 +73,14 @@ public Task<ActionResult> GetGcDump(int? pid)
             });
         }
 
-        [HttpGet("cpuprofile/{pid?}")]
-        public Task<ActionResult> CpuProfile(int? pid, [FromQuery][Range(-1, int.MaxValue)]int durationSeconds = 30)
-        {
-            TimeSpan duration = ConvertSecondsToTimeSpan(durationSeconds);
-            return InvokeService(async () =>
-            {
-                int pidValue = _diagnosticServices.ResolveProcess(pid);
-                IStreamWithCleanup result = await _diagnosticServices.StartCpuTrace(pidValue, duration, this.HttpContext.RequestAborted);
-                return new StreamWithCleanupResult(result, "application/octet-stream", FormattableString.Invariant($"{Guid.NewGuid()}.nettrace"));
-            });
-        }
-
         [HttpGet("trace/{pid?}")]
-        public Task<ActionResult> Trace(int? pid, [FromQuery][Range(-1, int.MaxValue)]int durationSeconds = 30)
+        public Task<ActionResult> Trace(int? pid, [FromQuery]TraceProfile profile = TraceProfile.All, [FromQuery][Range(-1, int.MaxValue)]int durationSeconds = 30)
```

Comment:
So does VS offer a good experience for this default? Does it show both events, perf graphs, (and soon metrics)?

Reply:
I'm only getting the CPU graph, CPU usage tab, and the Events tab.